Providers/nxOMSAutomationWorker/automationworker/3.x/scripts/onboarding3.py

```diff
@@ -0,0 +1,409 @@
+        test_mode               : bool,   test mode
+
+    Note:
+        The generated file has to match the latest worker.conf template.
+    """
+    worker_conf_path = os.path.join(state_directory_path, "worker.conf")
+
+    config = configparser.ConfigParser()
+    if os.path.isfile(worker_conf_path):
+        config.read(worker_conf_path)
+    conf_file = open(worker_conf_path, 'w')
+
+    worker_required_section = configuration3.WORKER_REQUIRED_CONFIG_SECTION
+    if not config.has_section(worker_required_section):
+        config.add_section(worker_required_section)
+    config.set(worker_required_section, configuration3.CERT_PATH, cert_path)
+    config.set(worker_required_section, configuration3.KEY_PATH, key_path)
+    config.set(worker_required_section, configuration3.BASE_URI, jrds_uri)
+    config.set(worker_required_section, configuration3.ACCOUNT_ID, automation_account_id)
+    config.set(worker_required_section, configuration3.MACHINE_ID, machine_id)
+    config.set(worker_required_section, configuration3.HYBRID_WORKER_GROUP_NAME, worker_group_name)
+    config.set(worker_required_section, configuration3.WORKING_DIRECTORY_PATH, working_directory_path)
+
+    worker_optional_section = configuration3.WORKER_OPTIONAL_CONFIG_SECTION
+    if not config.has_section(worker_optional_section):
+        config.add_section(worker_optional_section)
+    config.set(worker_optional_section, configuration3.PROXY_CONFIGURATION_PATH,
+               "/etc/opt/microsoft/omsagent/proxy.conf")
+    config.set(worker_optional_section, configuration3.STATE_DIRECTORY_PATH, state_directory_path)
+    if gpg_keyring_path is not None:
+        config.set(worker_optional_section, configuration3.GPG_PUBLIC_KEYRING_PATH, gpg_keyring_path)
+    if test_mode is True:
+        config.set(worker_optional_section, configuration3.BYPASS_CERTIFICATE_VERIFICATION, True)
+        config.set(worker_optional_section, configuration3.DEBUG_TRACES, True)
+
+    metadata_section = configuration3.METADATA_CONFIG_SECTION
+    if not config.has_section(metadata_section):
+        config.add_section(metadata_section)
+    config.set(metadata_section, configuration3.WORKER_TYPE, "diy")
+    config.set(metadata_section, configuration3.IS_AZURE_VM, str(is_azure_vm))
+    config.set(metadata_section, configuration3.VM_ID, vm_id)
+
+    registration_metadata_section = "registration-metadata"
+    if not config.has_section(registration_metadata_section):
+        config.add_section(registration_metadata_section)
+    config.set(registration_metadata_section, configuration3.REGISTRATION_ENDPOINT, registration_endpoint)
+    config.set(registration_metadata_section, configuration3.WORKSPACE_ID, workspace_id)
+    config.set(registration_metadata_section, configuration3.CERTIFICATE_THUMBPRINT, thumbprint)
+
+    config.write(conf_file)
+    conf_file.close()
+
+
+def get_autoregistered_worker_account_id():
```

Comment:
Test for system hybrid worker - Update Management

Reply:
This has registered the system hybrid worker and the hybrid worker for the system worker is running.